ProcessLib/LIE/HydroMechanics/HydroMechanicsProcess.cpp

```diff
@@ -568,13 +574,17 @@ void HydroMechanicsProcess<GlobalDim>::assembleWithJacobianConcreteProcess(
 {
     DBUG("AssembleWithJacobian HydroMechanicsProcess.");
 
+    const int process_id =
+        _use_monolithic_scheme ? 0 : _coupled_solutions->process_id;
```

Comment:
For another PR; I'd suggest to extract a function `currentProcessId()` into the base Process class.

Reply:
TODO once this PR is merged.